File: core/src/main/java/pcb/uwu/units/quantity/LuminousIntensityUnit.java
package pcb.uwu.units.quantity;

import pcb.uwu.core.BaseUnit;
import pcb.uwu.core.BigDecimalAmount;
import pcb.uwu.core.UnitCounter;

import java.util.function.Function;

public abstract class LuminousIntensityUnit extends BaseUnit {

	private final UnitCounter unitCounter;

	public LuminousIntensityUnit(String symbol,
								 String singularName,
								 String pluralName,
								 Function<BigDecimalAmount, BigDecimalAmount> translationToCanonical,
								 Function<BigDecimalAmount, BigDecimalAmount> translationFromCanonical) {

		super(6, symbol, singularName, pluralName, translationToCanonical, translationFromCanonical);

		this.unitCounter = new UnitCounter(this);
	}

	@Override
	public Class<? extends BaseUnit> getBaseUnitType() {
		return LuminousIntensityUnit.class;
	}

	@Override
	public UnitCounter getUnitCounter() {
		return unitCounter;
	}
}

File: core/src/main/java/pcb/uwu/unit/base/LuminousIntensityUnit.kt
package pcb.uwu.unit.base

import pcb.uwu.core.Amount
import pcb.uwu.core.BaseUnit
import pcb.uwu.core.UnitCounter
import java.util.function.Function

open class LuminousIntensityUnit(symbol: String,
                                 singularName: String,
                                 pluralName: String,
                                 translationToCanonical: Function<Amount, Amount>,
                                 translationFromCanonical: Function<Amount, Amount>) :
    BaseUnit(6, symbol, singularName, pluralName, translationToCanonical, translationFromCanonical)
{
    override val unitCounter
        get() = UnitCounter(this)
    override val baseUnitType = LuminousIntensityUnit::class.java
}